spring-cloud-gcp-data-spanner/src/main/java/com/google/cloud/spring/data/spanner/core/mapping/SpannerPersistentEntityImpl.java

```diff
@@ -407,70 +407,70 @@ public void setApplicationContext(ApplicationContext applicationContext)
 
 	@Override
 	public <B> PersistentPropertyAccessor<B> getPropertyAccessor(B object) {
-		PersistentPropertyAccessor<B> delegatedAccessor = super.getPropertyAccessor(object);
-		return new PersistentPropertyAccessor<B>() {
-
-			@Override
-			public void setProperty(PersistentProperty<?> property,
-					@Nullable Object value) {
-				if (property.isIdProperty()) {
-					SpannerPersistentEntity<?> owner = (SpannerPersistentEntity<?>) property.getOwner();
-					SpannerPersistentProperty[] primaryKeyProperties = owner.getPrimaryKeyProperties();
-
-					Iterator<Object> partsIterator;
-					if (value instanceof Key) {
-						Key keyValue = (Key) value;
-						if (keyValue.size() != primaryKeyProperties.length) {
-							throwWrongNumOfPartsException();
-						}
-						partsIterator = keyValue.getParts().iterator();
-					}
-					else {
-						if (primaryKeyProperties.length > 1) {
-							throwWrongNumOfPartsException();
-						}
-						partsIterator = Collections.singleton(value).iterator();
-					}
-					for (int i = 0; i < primaryKeyProperties.length; i++) {
-						SpannerPersistentProperty prop = primaryKeyProperties[i];
-						delegatedAccessor.setProperty(prop,
-								SpannerPersistentEntityImpl.this.spannerEntityProcessor.getReadConverter().convert(
-										partsIterator.next(), prop.getType()));
-					}
-				}
-				else {
-					delegatedAccessor.setProperty(property, value);
-				}
-			}
+		return new DelegatingPersistentPropertyAccessor<>(super.getPropertyAccessor(object));
+	}
 
-			private void throwWrongNumOfPartsException() {
-				throw new SpannerDataException(
-						"The number of key parts is not equal to the number of primary key properties");
+	@Override
+	public String getPrimaryKeyColumnName() {
+		SpannerPersistentProperty primaryKeyProperty = getPrimaryKeyProperties()[0];
+		return primaryKeyProperty.isEmbedded()
+				? this.spannerMappingContext.getPersistentEntityOrFail(primaryKeyProperty.getType()).getPrimaryKeyColumnName()
+				: primaryKeyProperty.getColumnName();
+	}
+
+	class DelegatingPersistentPropertyAccessor<B> implements PersistentPropertyAccessor<B> {
```

Comment:
Make this a `private static` class?

Reply:
Made it private, but it accesses a field in the outer class so it can't be static.

Reply:
Gotcha.